### ISSUE TITLE
Add libc++ support to steam-runtime

### ISSUE THREAD

**Issue by rtissera** [ValveSoftware/steam-runtime#27]
Hello,

I am working on porting a game to Linux using heavily C++11 features. As the game primary target compiler has been LLVM/Clang, I am glad to hear steam-runtime features clang 3.4.
However, clang3.4 does not always play very nice with libstdc++ bundled with GCC 4.8 for advanced C++11 features.
It would be really nice to see libc++ added to steam-runtime as it is the ideal companion for LLVM/Clang toolchain.

**Comment by baszalmstra** [comment]
+1

**Comment by wingman-j** [comment]
Yes please, the c++11/14 features are a really nice benefit of using clang!
